### ISSUE TITLE
[Filters] Hide 'More filters' sheet if there is only one filter

### ISSUE THREAD

**Issue by laurenmayers** [Shopify/polaris-react#2314]
## Issue summary
The [`Filters`](https://polaris.shopify.com/components/lists-and-tables/filters) component includes a 'More filters' button, which opens a sheet containing the full list of filters, including those displayed on the page as well as those that aren't. However, in cases where there is only one filter, this panel becomes redundant and may be confusing. 

## Expected behavior
The 'More filters' panel is only shown when there are more filters than those already rendered on the page, or more than one filter.

## Actual behavior
The 'More filters' panel is displayed regardless of how many filters there are and which filters are already displayed. 

## Steps to reproduce the problem
1. Add `<Filters>` to a component with one filter, and observe the 'More filters' panel.

## Reduced test case
This component with only one filter, `Account status`:
https://codesandbox.io/embed/festive-antonelli-3uc86

## Specifications
```bash
  System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 180.58 MB / 32.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.11.0 - ~/.nvm/versions/node/v10.11.0/bin/node
    Yarn: 1.12.3 - ~/.dev/yarn/1.12.3/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.11.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 77.0.3865.120
    Safari: 13.0.2
  npmPackages:
    @shopify/polaris: 4.5.0 => 4.5.0
    react: 16.9.0-alpha.0 => 16.9.0-alpha.0
    react-dom: 16.9.0-alpha.0 => 16.9.0-alpha.0
```

**Comment by manish-nx** [comment]
Any updates on this?
Apart from hiding the "More Filters" option through CSS, is there is any workaround for this?

**Comment by meecrobe** [comment]
@manish-nx this is fixed in 4.19.0 #2856
